src/libraries/System.Linq/src/System/Linq/Sum.cs

```diff
@@ -49,6 +72,101 @@ public static partial class Enumerable
             return sum;
         }
 
+        private static T SumSignedIntegersVectorized<T>(ReadOnlySpan<T> span)
+            where T : struct, IBinaryInteger<T>
+        {
+            Debug.Assert(span.Length >= Vector<T>.Count * 4);
```

Comment:
Could you also assert on some of the other checks that were made by the parent method? I.e. `Vector.IsHardwareAccelerated`/`IsSupported` etc?

Reply:
Done